### PR TITLE
MONGOID-5818 Support Rails 8

### PR DESCRIPTION
**Pull request by comandeo-mongo**

### CODE REVIEW

File: lib/mongoid/traversable.rb [hunk 1]
```diff
@@ -119,7 +123,7 @@ class << self
         # an existing field.
         # This condition also checks if the class has any descendants, because
         # if it doesn't then it doesn't need a discriminator key.
-        return unless !fields.key?(discriminator_key) && !descendants.empty?
+        return if fields.key?(discriminator_key) || descendants.empty?
```
Comment (jamis) [inline lib/mongoid/traversable.rb:126]: 💪 Love to see [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) in action :)

**Review by jamis — APPROVED** [review]
LGTM! That's actually fewer changes than I expected, after hearing what you'd run into earlier this week. Great work finding a way around those `class_attribute` changes!

**Comment by AlexKovynev** [comment]
8.0 released if so :)

**Comment by comandeo-mongo** [comment]
> 8.0 released if so :)

Oh cool, glad that we unblocked Rails team for the GA release! 😂

**Pull request merged**